src/bootstrap/util.rs

```diff
@@ -44,7 +44,13 @@ pub use t;
 /// Given an executable called `name`, return the filename for the
 /// executable for a particular target.
 pub fn exe(name: &str, target: TargetSelection) -> String {
-    if target.contains("windows") { format!("{}.exe", name) } else { name.to_string() }
+    if target.contains("windows") {
+        format!("{}.exe", name)
+    } else if target.contains("uefi") {
+        format!("{}.efi", name)
+    } else {
+        name.to_string()
+    }
```

Comment:
An option would be to use the output of `rustc --print file-names --crate-name {name} --crate-type bin`, but that might be a cyclic dependency if this function is used to find the path to the rustc executable.

Reply:
From what I see, this function is used with a lot of binaries instead of crates, so I don't think `rustc --print file-names --crate-name {name} --crate-type bin` can be used at all places where `exe` is currently being used.

Reply:
This feels a bit awkward regarding cyclic dependencies, doesn't it? Even if you fetch a stage from an external source, this stage will not reflect the last build, so this function will be out-of-sync with the source it builds. Not necessarily a big issue, but maybe requires a bit more investigation. Not sure it is worth blocking this change on it?

Reply:
Ideally you should ask the rustc binary that actually does the compilation for the artifact of which you want the filename. I do agree that it isn't worth blocking this change on.